packages/blockchain-wallet-v4-frontend/src/components/BuySell/Coinify/Create/AcceptTerms/index.js

```diff
@@ -0,0 +1,75 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+import { actions, model } from 'data'
+import AcceptTerms from './template'
+
+const { CHANGE } = model.components.coinify.REGISTER_STATES
+
+class AcceptTermsContainer extends PureComponent {
+  state = {
+    busy: false
+  }
+
+  componentDidUpdate (prevProps) {
+    if (this.props.signupError && !prevProps.signupError) {
+      // eslint-disable-next-line
+      this.setState({ busy: false })
+    }
+  }
+
+  handleResend = () => {
+    this.props.securityCenterActions.updateEmail(this.props.email)
+  }
+
+  onSubmit = () => {
+    this.setState({ busy: true })
```

Comment:
Is this a redux-form? Can you dispatch the submitting action for the form instead of adding this state and updating state inside componentDidUpdate?

Reply:
Good point. Component state is not necessary here.